tests/E2ETests/test_AlignmentReconstruction.py

```diff
@@ -3,25 +3,27 @@
 import shutil
 import numpy as np
 import unittest
-from pytom.basic.datatypes import DATATYPE_ALIGNMENT_RESULTS_RO, fmtAlignmentResultsRo as fmt, \
-    HEADER_ALIGNMENT_RESULTS_RO as header
-from pytom.basic.files import loadtxt, savetxt
 from pytom.agnostic.correlation import flcf, nxcc
 from pytom.agnostic.io import read
 from pytom.agnostic.tools import create_circle
-from numpy.fft import fftshift, fftn, ifftn
+from pytom.basic.datatypes import DATATYPE_ALIGNMENT_RESULTS_RO, fmtAlignmentResultsRo as fmt, \
+    HEADER_ALIGNMENT_RESULTS_RO as header
+from pytom.basic.files import loadtxt, savetxt
+from pytom.basic.transformations import general_transform2d
 from pytom.lib.pytom_volume import read as readC
 from pytom.lib.pytom_numpy import vol2npy
-from pytom.basic.transformations import general_transform2d
-
+from pytom.gpu.initialize import xp
```

Comment:
The problem with that is that stuff like `a = create_circle(data.shape, high, sigma)` on line 38 return a `cupy` array if this would import `cupy` (if the flags are correctly set). This will then break the test because you are using `cupy` arrays in `numpy` functions.

I was making sure to run test code in cupy mode by manually exporting `PYTOM_GPU=0` before running the unittest. Which triggers this behavior, but will also correctly import cupy on this line.

Reply:
Okay that sounds good. So from now these tests need to be run with explicitly exporting PYTOM_GPU=0?

Reply:
Only if you want to run them in `cupy`/`gpu` mode, without that it will run in `numpy`/`cpu`  mode